spec/test-outputs/www-integration.out.vcl

```diff
@@ -426,6 +459,43 @@ sub vcl_deliver {
   }
   # End dynamic section
 
+if (req.http.Cookie ~ "cookies_policy") {
+  if (req.http.Cookie:cookies_policy ~ "%22usage%22:true" && req.http.User-Agent !~ "^GOV\.UK Crawler Worker"){
+    if (table.lookup(active_ab_tests, "StartABusinessSegment") == "true") {
+      declare local var.variant STRING;
+      set var.variant = req.http.GOVUK-ABTest-StartABusinessSegment;
+
+      declare local var.sab_expiry TIME;
+      if (var.variant == "C") {
+        # If they have opted out of the banner or have visited the SaB checker
+        # Expire in 26 weeks (approx 6 months)
+        set var.sab_expiry = time.add(now, 26w);
+      } else {
+        # Otherwise, expire in half an hour, approx this session
+        set var.sab_expiry = time.add(now, 30m);
+      }
+
+      if (var.variant == "C") {
+        # If they have opted out of the banner or have visited the SaB checker
+        # Expire in 26 weeks (approx 6 months)
+        set var.sab_expiry = time.add(now, 26w);
+      } else if (req.http.GOVUK-ABTest-IsStartABusinessPage == "true" && var.variant == "A") {
+        # Otherwise, expire in half an hour, approx this session
+        set var.sab_expiry = time.add(now, 30m);
+        # If the user is in variant A and has seen a SaB page then we need to move them to B
+        # so they see the banner on the next SaB page they hit.
+        set var.variant = "B";
+      } else {
+        # Otherwise, expire in half an hour, approx this session
+        set var.sab_expiry = time.add(now, 30m);
```

Comment:
If `var.variant` is empty do we want to be setting this expiry at all? (disregard if you're planning to keep the previous block, as it doesn't matter you're setting it if it's forcing the set-cookie response header. I was just thinking of ways to tidy it up (using something like `if (var.variant ~ "(A|B|C))"` and then a dictionary look-up on `var.variant` to remove all of the ifs completely).

Reply:
Good shout - I think the new code is cleaner and yes we don't need to set this at all if the variant is empty :100: